### PR TITLE
More FPU stuff implemented

### PR DESCRIPTION
**Pull request by bebbo**

### CODE REVIEW

File: m68kconf.h [hunk 1]
```diff
@@ -108,7 +108,8 @@
 /* If ON, CPU will call the output reset callback when it encounters a reset
  * instruction.
  */
-#define M68K_EMULATE_RESET          OPT_OFF
+ /* CV Patch */
+#define M68K_EMULATE_RESET          OPT_ON
```
Comment (kstenerud) [inline m68kconf.h:112]: Does the default here need to be flipped? Also, does aline play nice when it's off?

File: m68kconf.h [hunk 2]
```diff
@@ -161,7 +162,8 @@
 /* If ON, CPU will call the instruction hook callback before every
  * instruction.
  */
-#define M68K_INSTRUCTION_HOOK       OPT_OFF
+ /* CV Patch */
+#define M68K_INSTRUCTION_HOOK       OPT_ON
```
Comment (kstenerud) [inline m68kconf.h:166]: Same: Does this need to be flipped?

File: m68kcpu.c [hunk 3]
```diff
@@ -597,6 +597,12 @@ static void default_instr_hook_callback(unsigned int pc)
 	(void)pc;
 }
 
+/* CV */
```
Comment (kstenerud) [inline m68kcpu.c:600]: Can these be taken out? Comments should only be there if they add important information.

**Review by kstenerud — COMMENTED** [review]
Looks good! Just a couple of small points.

**Comment by bebbo** [comment]
you should be able to edit the pull request... - otherwise I'll create a branch for this pull request

**Comment by bebbo** [comment]
The problem with your defines is: There is no convenient way to override these without change the headers.
You might consider something like
```
#ifndef XYZ_DEF
#define XYZ_DEF yeah!
#endif
```
then all defines can be configured in the Makefile.

**Comment by bebbo** [comment]
this way you could also add custom headers:
```
#ifdef INC_CUSTOM_HEADERS
#include "m68k-custom.h"
#endif
```
then one can define this and provide that custom header.
Or switch to c++17 which supports `has_include`...

**Comment by agentbooth** [comment]
At the end of the readme.txt, there's a note about using a custom conf -- would this solve your issue? I just recently learned of this myself.

> Using some custom m68kconf.h outside Musashi's directory
> --------------------------------------------------------
> It can be useful to keep an untouched musashi directory in a project (from
> git for example) and maintain a separate m68kconf.h specific to the
> project. For this, pass -DMUSASHI_CNF="mycustomconfig.h" to gcc (or whatever
> compiler you use). Notice that if you use an unix shell (or make which uses
> the shell to launch its commands), then you need to escape the quotes like
> this : -DMUSASHI_CNF=\"mycustomconfig.h\"
>

**Comment by kstenerud** [comment]
Yeah, the config values should be ifdefed for sure. I wrote this code long before I knew much about the preprocessor. I'll put that in later. But for now let's just keep the default config values as they are so that other people don't get weird surprises when they upgrade.

File: m68kcpu.c [hunk 4]
```diff
@@ -723,6 +729,30 @@ void m68k_set_reg(m68k_register_t regnum, unsigned int value)
...
+			return fx80_to_double(cpu->fpr[regnum]);
+	}
+	return 0;
+}
+
+int m68k_get_fpsr(void*context)
+{
+	return REG_FPSR;
+}
+
```
Comment (agentbooth) [inline m68kcpu.c:754]: Should use context? e.g.
```
m68ki_cpu_core* cpu = context != NULL ?(m68ki_cpu_core*)context : &m68ki_cpu;
return cpu->fpsr;
```

File: m68kconf.h [hunk 5]
```diff
@@ -175,6 +177,15 @@
  */
 #define M68K_EMULATE_ADDRESS_ERROR  OPT_OFF
 
+/* ---------- CV AddOns ---------- */
+
+/* Intercept illegal A-Line opcodes and call hook.
+   Hook prototype is: int hook()
+   If return == 0 then normal A-Line exception handling is performed.
+   Otherwise nothing is done.
```
Comment (agentbooth) [inline m68kconf.h:184]: should this be:
`if return == M68K_ALINE_EXCEPT`

File: m68kcpu.h [hunk 6]
```diff
@@ -683,6 +685,18 @@ extern jmp_buf m68ki_aerr_trap;
...
+#if M68K_ALINE_HOOK
+       #if M68K_ALINE_HOOK == OPT_SPECIFY_HANDLER
+               #define m68ki_aline_hook() M68K_ALINE_CALLBACK()
+       #else
+               #define m68ki_aline_hook() CALLBACK_ALINE_HOOK()
+       #endif
+#else
+       #define m68ki_aline_hook()  M68K_ALINE_EXCEPT
+#endif /* M68K_ALINE_HOOK */
+
```
Comment (agentbooth) [inline m68kcpu.h:699]: Do these need params like the other callbacks that take args?
e.g.
`#define m68ki_aline_hook(opcode,pc) M68K_ALINE_CALLBACK(opcode,pc)`

File: m68kcpu.h [hunk 7]
```diff
@@ -1946,6 +1960,10 @@ extern int cpu_log_enabled;
 /* Exception for A-Line instructions */
 static inline void m68ki_exception_1010(void)
 {
+#if M68K_ALINE_HOOK
+	int res = CALLBACK_ALINE_HOOK(REG_IR,ADDRESS_68K(REG_PPC));
```
Comment (agentbooth) [inline m68kcpu.h:1964]: Should this call be:
`	int res = m68ki_aline_hook(REG_IR,ADDRESS_68K(REG_PPC));`

**Pull request closed without merging**